### PR TITLE
Improve documentation and check* scripts 

### PR DESCRIPTION
**Pull request by Fenthick**
Issue: KNTK-352

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+docs/ linguist-generated=true
```
Comment (mmac-m3a) [inline .gitattributes:1]: Can we also add `go.mod` and `go.sum`?

File: tools/check_docs.sh [hunk 2]
```diff
@@ -6,19 +6,19 @@ source "tools/utility_functions.sh" || exit 1
...
-        git status --porcelain -- docs
+        echo "${changes}"
 
         echo -e "\nDiff for docs directory:"
         git diff -- docs
 
         echo -e "\nThis means that the documentation is not up-to-date."
-        echo "Please run 'make docs' and commit changes."
-        die
+        die "Please run 'make docs' and commit changes."
```
Comment (mmac-m3a) [inline tools/check_docs.sh:21]: `make docs` ran above. Would another run fix anything?

File: tools/check_go_mod.sh [hunk 3]
```diff
@@ -14,8 +14,7 @@ function run() {
         git diff -- go.mod go.sum
 
         echo -e "\nThis means that Go module definition is not consistent."
-        echo "Please run 'go mod tidy' and commit changes."
-        die
+        die "Please run 'go mod tidy' and commit changes."
```
Comment (mmac-m3a) [inline tools/check_go_mod.sh:17]: Similar here, will running `go mod tidy` again fix anything?

File: tools/utility_functions.sh [hunk 4]
```diff
@@ -2,36 +2,50 @@
...
     "$@"
 }
 
 function die() {
-    echo "Error - exit 1"
+    colored_echo RED "$*" 1>&2
     exit 1
 }
+
+function colored_echo() {
```
Comment (mmac-m3a) [inline tools/utility_functions.sh:28]: I like my name better :-)

**Review by mmac-m3a — APPROVED** [review]
Good to go in.

File: tools/check_docs.sh (hunk 2 above)
Comment (Fenthick) [reply tools/check_docs.sh:21]: No, but developer will see this text as the output from failed CI job, so he needs to run `make docs` locally.

File: tools/check_go_mod.sh (hunk 3 above)
Comment (Fenthick) [reply tools/check_go_mod.sh:17]: The same case as in https://github.com/kentik/terraform-provider-kentik-synthetics/pull/69#discussion_r765002422

**Pull request merged**